distcheck.cmake

```diff
@@ -46,38 +46,8 @@ MACRO(DISTCHECK_SETUP)
       find . -type d -print0 | xargs -0 chmod a-w
       && chmod u+w . && rm -rf _build _inst && mkdir -p _build && mkdir -p _inst
       && chmod u+rwx _build _inst && chmod a-w .
-      && cp ${CMAKE_BINARY_DIR}/CMakeCache.txt _build/
-      && ${SED} -i -e "'s|CMAKE_CACHEFILE_DIR:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_HOME_DIRECTORY:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_CXX_COMPILER:FILEPATH=.\\+||g'"
-                   -e "'s|CMAKE_CXX_FLAGS:STRING=.\\+||g'"
-                   -e "'s|CMAKE_CXX_FLAGS_DEBUG:STRING=.\\+||g'"
-                   -e "'s|CMAKE_CXX_FLAGS_MINSIZEREL:STRING=.\\+||g'"
-                   -e "'s|CMAKE_CXX_FLAGS_RELEASE:STRING=.\\+||g'"
-                   -e "'s|CMAKE_CXX_FLAGS_RELWITHDEBINFO:STRING=.\\+||g'"
-                   -e "'s|CMAKE_CXX_COMPILER-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_CXX_COMPILER_WORKS:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_CXX_FLAGS-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_CXX_FLAGS_DEBUG-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_CXX_FLAGS_MINSIZEREL-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_CXX_FLAGS_RELEASE-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_CXX_FLAGS_RELWITHDEBINFO-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_DETERMINE_CXX_ABI_COMPILED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_C_COMPILER:FILEPATH=.\\+||g'"
-                   -e "'s|CMAKE_C_FLAGS:STRING=.\\+||g'"
-                   -e "'s|CMAKE_C_FLAGS_DEBUG:STRING=.\\+||g'"
-                   -e "'s|CMAKE_C_FLAGS_MINSIZEREL:STRING=.\\+||g'"
-                   -e "'s|CMAKE_C_FLAGS_RELEASE:STRING=.\\+||g'"
-                   -e "'s|CMAKE_C_FLAGS_RELWITHDEBINFO:STRING=.\\+||g'"
-                   -e "'s|CMAKE_C_COMPILER-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_C_FLAGS-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_C_FLAGS_DEBUG-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_C_FLAGS_MINSIZEREL-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_C_FLAGS_RELEASE-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_C_FLAGS_RELWITHDEBINFO-ADVANCED:INTERNAL=.\\+||g'"
-                   -e "'s|CMAKE_DETERMINE_C_ABI_COMPILED:INTERNAL=.\\+||g'"
-                   _build/CMakeCache.txt
       && cd _build
+      && cmake ..
```

Comment:
I think this is very different with the original intent (and `cmake` is run anyway in the line below).

The benefit of the previous version is that it keeps the options that you enabled/disabled whereas this would erase any choice made.

Also, this still doesn't do anything on `APPLE` platform, is it desired?

Reply:
What I mean here is that the simple removing of cache data in the CMakeCache.txt is no more working.